velox/exec/Spiller.cpp

```diff
@@ -421,13 +429,18 @@ void Spiller::ensureSorted(SpillRun& run) {
   uint64_t sortTimeNs{0};
   {
     NanosecondTimer timer(&sortTimeNs);
-    gfx::timsort(
-        run.rows.begin(),
-        run.rows.end(),
-        [&](const char* left, const char* right) {
-          return container_->compareRows(
-                     left, right, state_.sortCompareFlags()) < 0;
-        });
+
+    if (!state_.prefixSortConfig().has_value()) {
+      PrefixSort::timSort(run.rows, container_, state_.sortCompareFlags());
+    } else {
+      PrefixSort::sort(
```

Comment:
PrefixSort needs additional memory for storing prefix. If we use it for spilling, will it cause more spills?

Reply:
We will reserve some memory for spill memory pool, if it exceeds the threshold, it may fail by OOM

Reply:
> PrefixSort needs additional memory for storing prefix. If we use it for spilling, will it cause more spills?

@zhli1142015 we allocate memory sort memory from spill memory pool which doesn't have quota enforcement. Also if we allocate from the spilling operator's memory pool and we do bypass the capacity check but there is a followup sanity check to make sure the spilling operator's memory usage drops after the spill finishes. Given that, we need to reserve some system memory for this sort of memory usage to prevent the potential server OOM risk. Also note the prefix sort memory is bounded by the sort run size which is configurable through spill config. cc @jinchengchenghh

Reply:
I see.